RestaurantReservation.API/Controllers/EmployeesController.cs

```diff
@@ -0,0 +1,194 @@
+    /// <returns>The employee data transfer object.</returns>
+    /// <response code="200">Returns the requested employee.</response>
+    /// <response code="404">If the employee with the specified ID is not found.</response>
+    [HttpGet("{employeeId}", Name = "GetEmployee")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(EmployeeDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetEmployee(int employeeId, bool includeOrders = false)
+    {
+        var employee = await _employeeRepository.GetEmployeeAsync(employeeId, includeOrders);
+        if (employee == null)
+        {
+            return NotFound(ApiErrors.EmployeeNotFound);
+        }
+
+        return Ok(includeOrders ? _mapper.Map<EmployeeWithOrdersDto>(employee) : _mapper.Map<EmployeeDto>(employee));
+    }
+
+    /// <summary>
+    /// Gets all managers.
+    /// </summary>
+    /// <returns>The list of managers.</returns>
+    /// <response code="200">Returns the list of managers.</response>
+    [HttpGet("managers")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<EmployeeDto>))]
+    public async Task<ActionResult<IEnumerable<EmployeeDto>>> GetManagers()
+    {
+        var employees = await _employeeRepository.GetManagers();
+        return Ok(_mapper.Map<IEnumerable<EmployeeDto>>(employees));
+    }
+
+    /// <summary>
+    /// Gets the average order amount for a specific employee.
+    /// </summary>
+    /// <param name="employeeId">The ID of the employee.</param>
+    /// <returns>The average order amount.</returns>
+    /// <response code="200">Returns the average order amount.</response>
+    /// <response code="404">If the employee with the specified ID is not found.</response>
+    [HttpGet("{employeeId}/average-order-amount")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(decimal?))]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<decimal?>> GetAverageOrderAmountForEmployee(int employeeId)
+    {
+        var averageOrderAmount = await _employeeRepository.GetAverageOrderAmountAsync(employeeId);
+
+        if (averageOrderAmount == null)
+        {
+            return NotFound(ApiErrors.EmployeeNotFound);
+        }
+
+        return Ok(new { AverageOrderAmount = averageOrderAmount });
+    }
+
+    /// <summary>
+    /// Creates a new employee.
+    /// </summary>
+    /// <param name="CraetedEmployee">The data for creating a new employee.</param>
+    /// <returns>The created employee.</returns>
+    /// <response code="201">Returns the created employee.</response>
+    /// <response code="400">If data is invalid.</response>
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status201Created, Type = typeof(EmployeeDto))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> PostEmployee(EmployeeCreationDto CraetedEmployee)
```

Comment:
Rename the method to **CreateEmployee**
plz avoid including the action type (PUT, POSt ..etc) in the api method name

the same applies for other controllers